spec/controllers/categories_controller_spec.rb

```diff
@@ -0,0 +1,218 @@
+# frozen_string_literal: true
+
+RSpec.describe CategoriesController, type: :controller do
+  let(:user) { create(:user1) }
+  let(:category) { create(:category, userid: user.id) }
+  let(:other_category) { create(:category, userid: user.id + 1) }
+  let(:valid_category_params) { attributes_for(:category).merge(userid: user.id) }
+  let(:invalid_category_params) { { name: nil, description: nil } }
+
+  describe 'GET #index' do
+    context 'when the user is logged in' do
```

Comment:
Given the meta-programming involved in setting @categories, it's probably worth verifying it is set correctly and that @page_tooltip is set.